llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUTargetStreamer.cpp

```diff
@@ -980,7 +980,12 @@ void AMDGPUTargetELFStreamer::EmitAmdhsaKernelDescriptor(
   if (KernelCodeSymbol->getVisibility() == ELF::STV_DEFAULT)
     KernelCodeSymbol->setVisibility(ELF::STV_PROTECTED);
 
+  Streamer.pushSection();
```

Comment:
This functions seems to be called during parser time. Otherwise, if it's only called at `finish` time`, pushSection/popSection would be unnecessary